File: 283. Move Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        read = 0
        write = 0
        while write < len(nums):
            if read < len(nums):
                if nums[read]:
                    nums[write] = nums[read]
                    write += 1
                    read += 1
                else:
                    read += 1
            else:
                nums[write] = 0
                write += 1

File: 283. Move Zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        if (nums==null || nums.length==0) return;
        int write = 0;
        for (int i=0; i<nums.length; i++) {
            if (nums[i]!=0) {
                nums[write++] = nums[i];
            }
        }
        while (write < nums.length) {
            nums[write++] = 0;
        }
    }
}